ophyd/_pyepics_shim.py

```diff
@@ -6,6 +6,16 @@
 
 from ._dispatch import _CallbackThread, EventDispatcher, wrap_callback
 
+import atexit
+
+
+def invalidate_ca():
+    global ca
+    ca = None
```

Comment:
May be it's a good idea to make a comment in the code on why exactly we want to invalidate `ca` (accidental attempt to use `ca` during interpreter shutdown will raise an exception, which is better than segfault, and we also check if `ca` is `None` when needed).

Reply:
Agreed, this definitely needs some comments to explain what's going on if we decide to go ahead with this.

As you correctly picked up, the basic idea is to make attempts to use `ca` "fail in a better way" so to speak. And to force the failure, instead of depending on seemingly random circumstances, like calling `pytest` with exactly the right arguments, etc.

One problem with this that we also import `epics`, `caget` and `caput` above. Even if we `None` all of them here, there are still references to `pv`s floating around and, for example, calls to `pv.clear_auto_monitor()` will trigger the problem.

So vigilance is still required and arguably picking on `ca` in particular is a bit arbitrary here.